lib/core/CJsonStateRestoreTraverser.cc

```diff
@@ -240,19 +240,21 @@ bool CJsonStateRestoreTraverser::parseNext(bool remember) {
 
         m_Handler.s_RememberValue = remember;
 
-        char c = '\0';
-        m_ReadStream.get(c);
-        if (c == '\0') {
-            break;
+        if (m_BytesRemaining == 0) {
+            m_BytesRemaining = BUFFER_SIZE - 1;
+            ::memset(m_Buffer, '\0', BUFFER_SIZE);
+            m_ReadStream.get(m_Buffer, BUFFER_SIZE, '\0');
```

Comment:
```suggestion
            ::memset(m_Buffer, '\0', BUFFER_SIZE);
            m_ReadStream.get(m_Buffer, BUFFER_SIZE, '\0');
            if (m_ReadStream.bad()) {
                // handle error
            }
            m_BytesRemaining = m_ReadStream.gcount();
```

What we're doing here is very similar to this: https://github.com/elastic/ml-cpp/blob/e7f6029489fc162b7461d30a26cc395dc82b895f/lib/api/CCsvInputParser.cc#L150-L168

That is tried and trusted code (used in the old Prelert Splunk app for years).